src/KOKKOS/kokkos_type.h

```diff
@@ -769,6 +769,38 @@ typedef tdual_virial_array::t_dev_um t_virial_array_um;
 typedef tdual_virial_array::t_dev_const_um t_virial_array_const_um;
 typedef tdual_virial_array::t_dev_const_randomread t_virial_array_randomread;
 
+// Spin Types
+
+//3d SP_FLOAT array n*4
+#ifdef LMP_KOKKOS_NO_LEGACY
+typedef Kokkos::DualView<X_FLOAT*[4], Kokkos::LayoutLeft, LMPDeviceType> tdual_sp_array;
+#else
+typedef Kokkos::DualView<X_FLOAT*[4], Kokkos::LayoutRight, LMPDeviceType> tdual_sp_array;
+#endif
+typedef tdual_sp_array::t_dev t_sp_array;
+typedef tdual_sp_array::t_dev_const t_sp_array_const;
+typedef tdual_sp_array::t_dev_um t_sp_array_um;
+typedef tdual_sp_array::t_dev_const_um t_sp_array_const_um;
+typedef tdual_sp_array::t_dev_const_randomread t_sp_array_randomread;
+
+//3d FM_FLOAT array n*3
+
+typedef Kokkos::DualView<F_FLOAT*[3], Kokkos::LayoutRight, LMPDeviceType> tdual_fm_array;
+typedef tdual_fm_array::t_dev t_fm_array;
+typedef tdual_fm_array::t_dev_const t_fm_array_const;
+typedef tdual_fm_array::t_dev_um t_fm_array_um;
+typedef tdual_fm_array::t_dev_const_um t_fm_array_const_um;
+typedef tdual_fm_array::t_dev_const_randomread t_fm_array_randomread;
+
+//3d FML_FLOAT array n*3
+
+typedef Kokkos::DualView<F_FLOAT*[3], Kokkos::LayoutRight, LMPDeviceType> tdual_fm_long_array;
```

Comment:
Please use the existing `tdual_f_array` instead of adding new types. You can typedef it to `tdual_fm_long_array` in the Spin class.